src/msac.rs

```diff
@@ -66,8 +67,8 @@ pub type ec_win = size_t;
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub struct MsacContext {
-    pub buf_pos: *const uint8_t,
-    pub buf_end: *const uint8_t,
+    buf_pos: *const uint8_t,
+    buf_end: *const uint8_t,
```

Comment:
Why store this as a start/end pointer combo? Why not store the length of the slice instead, since that's ultimately what you need to reconstruct it.

Reply:
Asm uses `MsacContext` so I can't change it at all.  Otherwise I'd definitely do something like this.